tests/ops/functions/test_bind_new_parameters.py

```diff
@@ -262,6 +310,8 @@ def test_linear_combination(H, new_coeffs, expected_H):
 
     assert qml.equal(new_H, expected_H)
     assert new_H is not H
+    assert all(no is not o for no, o in zip(new_H.operands, H.operands))
+    assert all(no is not o for no, o in zip(new_H.coeffs, H.coeffs))
```

Comment:
Optional suggestion: Not sure if this is needed anymore after the above discussion. Especially if any of the new/old coefficients are the same (or are `==1`), then it also doesn't make much sense to test this.
```suggestion
```